Generating baseline with Detekt 1.18.0-RC1 + Kotlin 1.5.1 uses all remaining disk then fails

<!-- The bug you're experiencing might have already be reported! -->
<!-- Please search in the [issues](https://github.com/detekt/detekt/issues) before creating one. -->

## Expected Behavior
<!-- Tell us what should happen -->
Generating baseline is currently using 100G+ of available disk space, until disk is full, then failing.
This happens with Detekt 1.8.0-RC1 and Kotlin 1.5.1.
It did not happen with Detekt 1.7.1 and Kotlin 1.4.32

## Observed Behavior
<!-- Tell us what happens instead of the expected behavior -->
```zsh
❯ cd ~/code/my_android_project

❯ du -hs .
 11G	.

❯ ./gradlew detektBaselineMain detektBaselineTest
...
...
java.io.IOException: No space left on device
...
BUILD FAILED in 2h 47m 55s
16407 actionable tasks: 10060 executed, 5100 from cache, 1247 up-to-date
Publishing build scan...
The build scan publishing request was rejected as the data exceeded the maximum size.

❯ du -hs .
114G	.

❯ ./gradlew clean cleanBuildCache
...

❯ du -hs .
 11G	.
```

## Steps to Reproduce
<!-- Provide a link to a live example, or an unambiguous set of steps to -->
<!-- reproduce this bug. Include code to reproduce, if relevant -->
This is a large android project (~1M lines of Kotlin, 300+ modules, and 20 product flavors), and we use detekt tasks that include type resolution.

## Context
<!-- How has this issue affected you? What are you trying to accomplish? -->
<!-- Providing context helps us come up with a solution that is most useful in the real world -->
We use the gradle plugin version of detekt.

Running `detektMain detektTest` tasks succeed.
Running `detektBaselineMain detektBaselineTest` task fail.

I've tried adding the following config to reduce the number of targets, but get the same issue (uses up all remaining disk space the fails):
```
      ignoredBuildTypes = ["release"]
      ignoredFlavors = [// all our flavors except for 1]
```

We have the following config:
- parallel = true
- allRules = true
- jvmTarget = "1.8"
- `classpath.setFrom(project.configurations.getByName("detekt"))` on both Detekt and DetektCreateBaselineTask tasks.

## Your Environment
<!-- Include as many relevant details about the environment you experienced the bug in -->
* Version of detekt used: 1.8.0-RC1
* Version of Gradle used (if applicable): Gradle 7.0.2
* Gradle scan link (add `--scan` option when running the gradle task): Scan fails with "The build scan publishing request was rejected as the data exceeded the maximum size." error.
* Operating System and version: macOS Catalina (v10.15.7)
* Link to your project (if it's a public repository): N/A - not open source

--- a/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/internal/DetektAndroid.kt
+++ b/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/internal/DetektAndroid.kt
@@ -16,7 +16,6 @@ import org.gradle.api.Project
 import org.gradle.api.Task
 import org.gradle.api.file.FileCollection
 import org.gradle.api.tasks.TaskProvider
-import org.gradle.jvm.tasks.Jar
 import java.io.File
 
 internal class DetektAndroid(private val project: Project) {
@@ -75,43 +74,24 @@ internal class DetektAndroid(private val project: Project) {
                 baseExtension.variants
                     ?.matching { !extension.matchesIgnoredConfiguration(it) }
                     ?.all { variant ->
-                        val intermediatesJavacJar = project.registerIntermediatesJavacJarTask(variant)
-                        project.registerAndroidDetektTask(
-                            bootClasspath = bootClasspath,
-                            extension = extension,
-                            variant = variant,
-                            intermediatesJavaCJar = intermediatesJavacJar
-                        ).also { provider ->
+                        project.registerAndroidDetektTask(bootClasspath, extension, variant).also { provider ->
                             mainTaskProvider.dependsOn(provider)
                         }
-                        project.registerAndroidCreateBaselineTask(
-                            bootClasspath = bootClasspath,
-                            extension = extension,
-                            variant = variant,
-                            intermediatesJavaCJar = intermediatesJavacJar
-                        ).also { provider ->
-                            mainBaselineTaskProvider.dependsOn(provider)
-                        }
+                        project.registerAndroidCreateBaselineTask(bootClasspath, extension, variant)
+                            .also { provider ->
+                                mainBaselineTaskProvider.dependsOn(provider)
+                            }
                         variant.testVariants
                             .filter { !extension.matchesIgnoredConfiguration(it) }
                             .forEach { testVariant ->
-                                val testIntermediatesJavacJar = project.registerIntermediatesJavacJarTask(testVariant)
-                                project.registerAndroidDetektTask(
-                                    bootClasspath = bootClasspath,
-                                    extension = extension,
-                                    variant = testVariant,
-                                    intermediatesJavaCJar = testIntermediatesJavacJar
-                                ).also { provider ->
-                                    testTaskProvider.dependsOn(provider)
-                                }
-                                project.registerAndroidCreateBaselineTask(
-                                    bootClasspath,
-                                    extension,
-                                    testVariant,
-                                    testIntermediatesJavacJar
-                                ).also { provider ->
-                                    testBaselineTaskProvider.dependsOn(provider)
-                                }
+                                project.registerAndroidDetektTask(bootClasspath, extension, testVariant)
+                                    .also { provider ->
+                                        testTaskProvider.dependsOn(provider)
+                                    }
+                                project.registerAndroidCreateBaselineTask(bootClasspath, extension, testVariant)
+                                    .also { provider ->
+                                        testBaselineTaskProvider.dependsOn(provider)
+                                    }
                             }
                     }
             }
@@ -126,17 +106,11 @@ internal class DetektAndroid(private val project: Project) {
     private fun Project.registerAndroidDetektTask(
         bootClasspath: FileCollection,
         extension: DetektExtension,
-        variant: BaseVariant,
-        intermediatesJavaCJar: TaskProvider<Jar>,
+        variant: BaseVariant
     ): TaskProvider<Detekt> =
         registerDetektTask(DetektPlugin.DETEKT_TASK_NAME + variant.name.capitalize(), extension) {
             setSource(variant.sourceSets.map { it.javaDirectories })
-            classpath.apply {
-                setFrom(variant.getCompileClasspath(null).filter { it.exists() })
-                from(bootClasspath)
-                from(intermediatesJavaCJar.map { it.destinationDirectory.asFileTree })
-            }
-            dependsOn(intermediatesJavaCJar)
+            classpath.setFrom(variant.getCompileClasspath(null).filter { it.exists() } + bootClasspath)
             // If a baseline file is configured as input file, it must exist to be configured, otherwise the task fails.
             // We try to find the configured baseline or alternatively a specific variant matching this task.
             extension.baseline?.existingVariantOrBaseFile(variant.name)?.let { baselineFile ->
@@ -153,36 +127,13 @@ internal class DetektAndroid(private val project: Project) {
     private fun Project.registerAndroidCreateBaselineTask(
         bootClasspath: FileCollection,
         extension: DetektExtension,
-        variant: BaseVariant,
-        intermediatesJavaCJar: TaskProvider<Jar>,
+        variant: BaseVariant
     ): TaskProvider<DetektCreateBaselineTask> =
         registerCreateBaselineTask(DetektPlugin.BASELINE_TASK_NAME + variant.name.capitalize(), extension) {
             setSource(variant.sourceSets.map { it.javaDirectories })
-            classpath.apply {
-                setFrom(variant.getCompileClasspath(null).filter { it.exists() })
-                from(bootClasspath)
-                from(intermediatesJavaCJar.map { it.destinationDirectory.asFileTree })
-            }
-            dependsOn(intermediatesJavaCJar)
+            classpath.setFrom(variant.getCompileClasspath(null).filter { it.exists() } + bootClasspath)
             val variantBaselineFile = extension.baseline?.addVariantName(variant.name)
             baseline.set(project.layout.file(project.provider { variantBaselineFile }))
             description = "EXPERIMENTAL: Creates detekt baseline for ${variant.name} classes with type resolution"
         }
-
-    private fun Project.registerIntermediatesJavacJarTask(variant: BaseVariant): TaskProvider<Jar> {
-        val jarTaskName = "detektIntermediatesJavacJar${variant.name.capitalize()}"
-        val jarOutputDir = layout.buildDirectory.dir(jarTaskName).get()
-        return tasks.register(jarTaskName, Jar::class.java) { task ->
-            task.from(
-                layout
-                    .buildDirectory
-                    .dir("intermediates/javac/${variant.name}/classes")
-                    .get()
-                    .asFileTree
-                    .matching { it.include("**/*.class") }
-            )
-            task.destinationDirectory.set(jarOutputDir)
-            task.dependsOn(variant.assembleProvider)
-        }
-    }
 }

--- a/detekt-gradle-plugin/src/test/kotlin/io/gitlab/arturbosch/detekt/DetektAndroidSpec.kt
+++ b/detekt-gradle-plugin/src/test/kotlin/io/gitlab/arturbosch/detekt/DetektAndroidSpec.kt
@@ -378,43 +378,6 @@ object DetektAndroidSpec : Spek({
                 }
             }
         }
-
-        describe("android tasks have javac intermediates on classpath") {
-            val projectLayout = ProjectLayout(
-                numberOfSourceFilesInRootPerSourceDir = 0,
-            ).apply {
-                addSubmodule(
-                    name = "app",
-                    numberOfSourceFilesPerSourceDir = 0,
-                    numberOfCodeSmells = 0,
-                    buildFileContent = """
-                        $APP_PLUGIN_BLOCK
-                        $ANDROID_BLOCK_WITH_VIEW_BINDING
-                        $DETEKT_BLOCK
-                    """.trimIndent(),
-                    srcDirs = listOf("src/main/java"),
-                )
-            }
-            val gradleRunner = createGradleRunnerAndSetupProject(projectLayout, dryRun = false)
-
-            gradleRunner.projectFile("app/src/main/java").mkdirs()
-            gradleRunner.projectFile("app/src/main/res/layout").mkdirs()
-            gradleRunner.writeProjectFile("app/src/main/AndroidManifest.xml", MANIFEST_CONTENT)
-            gradleRunner.writeProjectFile("app/src/main/res/layout/activity_sample.xml", SAMPLE_ACTIVITY_LAYOUT)
-            gradleRunner.writeProjectFile("app/src/main/java/SampleActivity.kt", SAMPLE_ACTIVITY_USING_VIEW_BINDING)
-
-            it("task :app:detektMain has javac intermediates on the classpath") {
-                gradleRunner.runTasksAndCheckResult(":app:detektMain") { buildResult ->
-                    assertThat(buildResult.output).doesNotContain("error: unresolved reference: databinding")
-                }
-            }
-
-            it("task :app:detektTest has javac intermediates on the classpath") {
-                gradleRunner.runTasksAndCheckResult(":app:detektTest") { buildResult ->
-                    assertThat(buildResult.output).doesNotContain("error: unresolved reference: databinding")
-                }
-            }
-        }
     }
 })
 
@@ -476,19 +439,6 @@ private val ANDROID_BLOCK_WITH_FLAVOR = """
     }
 """.trimIndent()
 
-private val ANDROID_BLOCK_WITH_VIEW_BINDING = """
-    android {
-        compileSdkVersion = 30
-        defaultConfig {
-            applicationId = "io.gitlab.arturbosch.detekt.app"
-            minSdkVersion = 24
-        }
-        buildFeatures {
-            viewBinding = true
-        }
-    }
-""".trimIndent()
-
 private val DETEKT_BLOCK = """
     detekt {
         reports {
@@ -497,39 +447,7 @@ private val DETEKT_BLOCK = """
     }
 """.trimIndent()
 
-private val SAMPLE_ACTIVITY_LAYOUT = """
-    <?xml version="1.0" encoding="utf-8"?>
-    <View
-        xmlns:android="http://schemas.android.com/apk/res/android"
-        android:id="@+id/sample_view"
-        android:layout_width="match_parent"
-        android:layout_height="match_parent"
-        />
-""".trimIndent()
-
-private val SAMPLE_ACTIVITY_USING_VIEW_BINDING = """
-    package io.gitlab.arturbosch.detekt.app
-    
-    import android.app.Activity
-    import android.os.Bundle
-    import android.view.LayoutInflater
-    import io.gitlab.arturbosch.detekt.app.databinding.ActivitySampleBinding
-    
-    class SampleActivity : Activity() {
-    
-        private lateinit var binding: ActivitySampleBinding
-    
-        override fun onCreate(savedInstanceState: Bundle?) {
-            binding = ActivitySampleBinding.inflate(LayoutInflater.from(this))
-            setContentView(binding.root)
-        }
-    }
-""".trimIndent() + "\n" // new line at end of file rule
-
-private fun createGradleRunnerAndSetupProject(
-    projectLayout: ProjectLayout,
-    dryRun: Boolean = true,
-) = DslGradleRunner(
+private fun createGradleRunnerAndSetupProject(projectLayout: ProjectLayout) = DslGradleRunner(
     projectLayout = projectLayout,
     buildFileName = "build.gradle",
     mainBuildFileContent = """
@@ -541,5 +459,5 @@ private fun createGradleRunnerAndSetupProject(
             }
         }
     """.trimIndent(),
-    dryRun = dryRun,
+    dryRun = true
 ).also { it.setupProject() }

